### PR TITLE
shelldap: 1.4.0 -> 1.5.1

### PR DESCRIPTION
**Pull request by clerie**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/shelldap/default.nix [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
     touch Makefile.PL
   '';
+
   installPhase = ''
     runHook preInstall
     install -Dm555 -t $out/bin shelldap
     runHook preInstall
   '';
+
+  doCheck = false;
```
Comment (Janik-Haag) [inline pkgs/tools/misc/shelldap/default.nix:38]: please add a comment on why checks are disabled

**Comment by tobiasBora** [comment]
This seems good to me once you explain why checks are disabled. But I can't really test anything as I don't have access to an ldap server anymore, so I trust you on the fact that it works! Anyway, the current derivation certainly fails as the gitlab server seems to have moved to github.

**Comment by clerie** [comment]
@Janik-Haag @tobiasBora Thank you for your review, I added a comment to `doCheck = false`. If you want to try this tool, there are plenty of public LDAP servers to connect to.

**Review by yu-re-ka — CHANGES_REQUESTED** [review]
Ofborg is reporting a src hash mismatch

```
trying https://github.com/mahlonsmith/shelldap/archive/refs/tags/v1.5.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 26030    0 26030    0     0   147k      0 --:--:-- --:--:-- --:--:--  147k
unpacking source archive /build/v1.5.1.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/b1wa8hhbrclbvwysdz40qdw8ffady4wz-source.drv':
         specified: sha256-xaizHuf/0x32UY3lQFYmI7EKxHRO72aoI9IuKj6yqcQ=
            got:    sha256-67ttAXzu9pfeqjfhMfLMb9vWCXTrE+iUDCbamqswaLg=
error: 1 dependencies of derivation '/nix/store/4ibv0zgmjd9qamhp8v2lnb61s8majsfl-perl5.38.0-shelldap-1.5.1.drv' failed to build
```

locally I can confirm this:

```
[yuka@m1:~]$ nix-prefetch-github mahlonsmith shelldap --rev v1.5.1
{
    "owner": "mahlonsmith",
    "repo": "shelldap",
    "rev": "54885085b8ec0b8063f4144bfa80ba4b72698f39",
    "hash": "sha256-67ttAXzu9pfeqjfhMfLMb9vWCXTrE+iUDCbamqswaLg="
}

[yuka@m1:~]$ 

```

**Comment by clerie** [comment]
@yu-re-ka oh thanks for your review. I found out what wen't wrong. When I initially changed the upstream repo, I forgot to bump the version. I noticed that und changed the version in the derivation. The build was successful, as the unchanged source hash was still cached and it actually build an old version.

I updated the hash now and had to update the dependencies too. It should work now.

**Comment by yu-re-ka** [comment]
@ofborg build shelldap

File: pkgs/tools/misc/shelldap/default.nix [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+
   meta = with lib; {
-    homepage = "https://bitbucket.org/mahlon/shelldap/";
+    homepage = "https://github.com/mahlonsmith/shelldap/";
     description = "A handy shell-like interface for browsing LDAP servers and editing their content";
+    changelog = "https://github.com/mahlonsmith/shelldap/blob/v${version}/CHANGELOG";
     license = with licenses; [ bsd3 ];
-    maintainers = with maintainers; [ tobiasBora ];
+    maintainers = with maintainers; [ clerie tobiasBora ];
     platforms = platforms.linux;
```
Comment (yu-re-ka) [inline pkgs/tools/misc/shelldap/default.nix:50]: What makes this linux-only? The readme lists several BSDs and Mac OS, so `platforms.unix`.

Comment (clerie) [reply pkgs/tools/misc/shelldap/default.nix:50]: In the initial PR the Darwin build was failing:
https://github.com/NixOS/nixpkgs/pull/56933#issuecomment-471206963

So builds got restricted to Linux only:
https://github.com/NixOS/nixpkgs/pull/56933#issuecomment-499298699

I'm not able to test this, but I think it is worth trying building it out on Darwin again.

Comment (yu-re-ka) [reply pkgs/tools/misc/shelldap/default.nix:50]: Yeah, let's try it. If ofborg can build darwin why not include it, and if not, set `broken = stdenv.isDarwin;`

**Review by yu-re-ka — APPROVED** [review]

**Comment by yu-re-ka** [comment]
Folowup: https://github.com/NixOS/nixpkgs/pull/258079

**Pull request merged**